### PR TITLE
refine dangling checks

### PR DESCRIPTION
**Pull request by vrothberg**
By proxy by vendoring containers/common. Previously, a "dangling" image
was an untagged image; just a described in the Docker docs. The
definition of dangling has now been refined to an untagged image without
children to be compatible with Docker.

Further update a redundant image-prune test.

Fixes: #10998
Fixes: #10832
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
@containers/podman-maintainers PTAL

File: go.mod [hunk 1]
```diff
@@ -69,3 +69,5 @@ require (
 	k8s.io/api v0.21.3
 	k8s.io/apimachinery v0.21.3
 )
+
+replace github.com/containers/common => github.com/vrothberg/common v0.0.3-0.20210720091711-39e7214a081e
```
Comment (rhatdan) [inline go.mod:73]: Looks like this is still a WIP?

File: pkg/domain/infra/abi/images_list.go [hunk 2]
```diff
@@ -30,12 +30,16 @@ func (ir *ImageEngine) List(ctx context.Context, opts entities.ImageListOptions)
 		for j, d := range img.Digests() {
 			digests[j] = string(d)
 		}
+		isDangling, err := img.IsDangling(ctx)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error checking if image %q is dangling", img.ID())
```
Comment (rhatdan) [inline pkg/domain/infra/abi/images_list.go:35]: NIT: I don't think we should start errors with "error", Podman prepends Error: on all Errors, I don't want to end up with 
Error: error checking if image ...

File: go.mod (hunk 1 above)
Comment (vrothberg) [reply go.mod:73]: Thank you for catching that!  I missed to remove that line.

File: pkg/domain/infra/abi/images_list.go (hunk 2 above)
Comment (vrothberg) [reply pkg/domain/infra/abi/images_list.go:35]: I'll tackle that in a separate PR.  There more such cases in the code that I can tackle at once.

**Comment by rhatdan** [comment]
LGTM

**Comment by vrothberg** [comment]
@containers/podman-maintainers @Luap99 PTAL

**Comment by TomSweeneyRedHat** [comment]
Does `podman images --filter dangling=true` work appropriately with these changes?

Also the man page for `podman rmi` defines an untagged image as a dangling one, does that man page need changing?

**Comment by vrothberg** [comment]
> Does `podman images --filter dangling=true` work appropriately with these changes?

Yes.

> Also the man page for `podman rmi` defines an untagged image as a dangling one, does that man page need changing?

Good catch, thanks!  Fixed :+1:

**Comment by vrothberg** [comment]
@containers/podman-maintainers PTAL

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM
and happy green test buttons.

**Comment by vrothberg** [comment]
merge me :)

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**